stl/inc/array

```diff
@@ -617,9 +617,13 @@ public:
     _DEPRECATE_TR1_NAMESPACE void assign(const _Ty&) {}
 #endif // _HAS_TR1_NAMESPACE
 
-    _CONSTEXPR20 void fill(const _Ty&) {}
+    _CONSTEXPR20 void fill(const _Ty&) {
+        static_assert(!is_const_v<_Ty>, "cannot fill zero-sized array of type <const T>");
+    }
 
-    _CONSTEXPR20 void swap(array&) noexcept {}
+    _CONSTEXPR20 void swap(array&) noexcept {

```

Comment:
I believe this is actually not the correct fix.

According to the [standard](http://eel.is/c++draft/concept.swappable#3) `std::array<const T, 0>` would fulfill the `swappable` concept, as that does not check the body of the function.

I think the correct thing would be to use SFINAE
```suggestion
    template<class _OTy = _Ty, enable_if_t<is_const_v<_OTy>, int> = 0>
    _CONSTEXPR20 void swap(array&) noexcept {
```

Reply:
Note that we should also add some tests for this somewhere (Do we include concepts in array?)
```cpp
#if __cpp_lib_concepts
static_assert(swappable<array<int, 0>);
static_assert(!swappable<array<const int, 0>);
#endif // __cpp_lib_concepts
```

Reply:
I've applied your suggestion. Although I'm not sure using SFINAE here won't break the ABI

Reply:
I wouldnt think so, as there is nothing virtual here and we can always add member functions. 

The one thing that could happen is that (buggy / ill formed) user code that relies on this function would break

Reply:
`swappable` and `is_swappable_v` care about non-member `swap` and ignore member `swap`; if you look at e.g. the Standard containers, you'll notice that non-member `swap` generally has SFINAE to make the trait and concept work, but member `swap` does not so it can provide nice clear error messages.